sdk/python/lib/pulumi/provider/experimental/ComponentProvider.md

```diff
@@ -0,0 +1,185 @@
+
+At this point, we can run the provider, but it does not yet do anything useful. To get data in and out of the provider we need to define the inputs and outputs of the component. Inputs are infered from the `args` parameter of the component's constructor. Add a new type `MyComponentArgs` derived from `TypedDict` to represent the arguments, and add it as an argument to the constructor.
+
+```diff
+-from typing import Optional
++from typing import Optional, TypedDict
+
+ import pulumi
+
++class MyComponentArgs(TypedDict):
++    who: Optional[pulumi.Output[str]]
++
+ class MyComponent(pulumi.ComponentResource):
+-    def __init__(self, name: str, opts: Optional[pulumi.ResourceOptions] = None):
++    def __init__(self, name: str, args: MyComponentArgs, opts: Optional[pulumi.ResourceOptions] = None):
+         super().__init__("my-provider:index:MyComponent", name, {}, opts)
+         self.register_outputs({})
+```
+
+Outputs are defined as attributes on the component class:
+
+```diff
+ class MyComponent(pulumi.ComponentResource):
++    greeting: Optional[pulumi.Output[str]]
++
+     def __init__(self, name: str, args: MyComponentArgs, opts: Optional[pulumi.ResourceOptions] = None):
+         super().__init__("my-provider:index:MyComponent", name, {}, opts)
+         self.register_outputs({})
+```
+
+To return data from the component, assign a value to the output attribute:
+
+```diff
+     def __init__(self, name: str, args: MyComponentArgs, opts: Optional[pulumi.ResourceOptions] = None):
+         super().__init__("my-provider:index:MyComponent", name, {}, opts)
++        who = args.get("who") or "Pulumipus"
++        self.greeting = pulumi.Output.concat("Hello, ", who, "!")
+         self.register_outputs({})
+```
+
+With everything in place, we can now run the provider, and request the schema:
+
+```bash
+# Create a venv and install the dependencies
+python -m venv venv
+source venv/*/activate
+pip install -r requirements.txt
+
+# Get the schema
+pulumi package get-schema ./
+```
+
+```json
+{
+  "name": "my-provider",
+  "displayName": "My Component Provider",
+  "version": "1.2.3",
+  "meta": {
+    "moduleFormat": "(.*)"
+  },
+  "language": { ... },
+  "config": {},
+  "provider": {
+    "type": "object"
+  },
+  "resources": {
+    "my-provider:index:MyComponent": {
+      "properties": {
+        "greeting": {
+          "type": "string"
+        }
+      },
+      "type": "object",
+      "required": [
+        "greeting"
+      ],
+      "inputProperties": {
+        "who": {
+          "type": "string"
+        }
+      },
+      "isComponent": true
+    }
+  }
+}
+```
+
+To use the provider from a local project, create a new directory and add a `Pulumi.yaml` file with a reference to the provider under the `plugins.provider` key:
```

Comment:
A new directory where? You imply a subfolder in the provider folder, maybe spell that out?